climpred/prediction.py

```diff
@@ -38,19 +48,72 @@ def compute_perfect_model(
         metric (str): `metric` name, see :py:func:`climpred.utils.get_metric_function`.
         comparison (str): `comparison` name, see
                           :py:func:`climpred.utils.get_comparison_function`.
+        dim (str or list): dimension to apply metric over. default: ['member', 'init']
         add_attrs (bool): write climpred compute args to attrs. default: True
+        ** metric_kwargs (dict): additional keywords to be passed to metric
 
     Returns:
-        skill (xarray object): skill score.
+        skill (xarray object): skill score with dimensions as input `ds`
+                               without `dim`.
 
     """
-    supervector_dim = 'svd'
-    metric = get_metric_function(metric, PM_METRICS)
+    # if stack, comparisons return forecast with member dim and reference
+    # without member dim which is needed for probabilistic
+    # if not stack, comparisons return forecast and reference with member dim
+    # which is neeeded for deterministic
+    if metric in PROBABILISTIC_METRICS:
+        if comparison not in PROBABILISTIC_PM_COMPARISONS:
+            raise ValueError(
+                f'Probabilistic metric {metric} cannot work with comparison \
+                 {comparison}'
+            )
+        stack = False
+        # dim = 'member'
+    elif set(dim) == set(['init', 'member']):
+        dim_to_apply_metric_to = 'svd'
+        stack = True
+    else:
+        if metric in ['pr', 'pearson_r', 'acc']:
```

Comment:
We should probably have a dictionary defining all the aliases

Reply:
now check in the beginning of the compute function